TestHelper.psm1

```diff
@@ -201,3 +204,210 @@ function Install-ResourceDesigner {
     
     return (Get-Module -Name $Script:DesignerModuleName -ListAvailable)
 }
+
+
+<#
+    .SYNOPSIS
+        Initializes an enviroment for running unit or integration tests
+        on a DSC resource.
+        
+        This includes the following things:
+        1. Creates a temporary working folder.
+        2. Updates the $env:PSModulePath to ensure the correct module is tested.
+        3. Backs up any settings that need to be changed to accurately test
+           the resource.
+        4. Produces a test object containing any parameters that may be used
+           for testing as well as storing the backed up settings.
+           
+        The above changes are reverted by calling the Restore-TestEnvironment
+        function. This includes deleteing the temporary working folder.
+    
+    .PARAMETER DSCModuleName
+        The name of the DSC Module containing the resource that the tests will be
+        run on.
+
+    .PARAMETER DSCResourceName
+        The full name of the DSC resource that the tests will be run on. This is
+        usually the name of the folder containing the actual resource MOF file.
+
+    .PARAMETER TestType
+        Specifies the type of tests that are being intialized. It can be:
+        Unit: Initialize for running Unit tests on a DSC resource. Default.
+        Integration: Initialize for running Integration tests on a DSC resource.
+
+    .OUTPUT
+        Returns a test environment object which must be passed to the
+        Restore-TestEnvironment function to allow it to restore the system
+        back to the original state as well as clean up and working/temp files.
+        
+    .EXAMPLE
+        $TestEnvironment = Inialize-TestEnvironment `
+            -DSCModuleName 'xNetworking' `
+            -DSCResourceName 'MSFT_xFirewall'
+            
```

Comment:
this example lacks -TestType Unit
We could make it default value, but I think it's better to require user to specify it explicitly so that it's clear that this function works differently for different test types.